keps/349-all-or-nothing/README.md

```diff
@@ -0,0 +1,425 @@
+all pods to be running at the same time, would not deadlock when scheduling
+their pods. This could happen in case of node provisioning issues to match
+the configured cluster queue quota and when the Jobs don't specify priorities
+(or specify the same priority).
+
+My use case can be supported by setting `waitForPodsReady=true` in the Kueue
+configuration.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+If a workload fails to schedule its pods it could block admission of other
+workloads indefinitely.
+
+To mitigate this issue we introduce a timeout on reaching the `PodsReady`
+condition by a workload since its job start (see:
+[Timeout on reaching the PodsReady condition](#timeout-on-reaching-the-podsready-condition)).
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Kueue Configuration API
+
+We extend the global Kueue Configuration API to introduce the new fields:
+`waitForPodsReady` to opt-in for the new behavior, and `podsReadyTimeout` to
+timeout if a workload takes too long to reach the `PodsReady` condition.
+
+```golang
+// Configuration is the Schema for the kueueconfigurations API
+type Configuration struct {
+  ...
+  // waitForPodsReady, when true, indicates that each admitted workload
+  // blocks admission of other workloads in the cluster, until it is in the
+  // `PodsReady` condition. If false, all workloads start as soon as they are
+  // admitted and do not block admission of other workloads. The PodsReady
+  // condition is only added if this setting is enabled. If unspecified,
+  // it defaults to false.
+  waitForPodsReady *bool `json:"waitForPodsReady,omitempty"`
+
+  // podsReadyTimeout describes the timeout for an admitted workload to reach
+  // the PodsReady condition since unsuspending of the corresponding Job.
+  // After exceeding this timeout the corresponding job gets suspended again
+  // and moved to the ClusterQueue's inadmissibleWorkloads list. The timeout is
+  // enforced only if waitForPodsReady=true. If unspecified, it defaults to 5min.
+  podsReadyTimeout *metav1.Time `json:"podsReadyTimeout,omitempty"`
```

Comment:
```yaml
waitForPodsReady:
  enabled: true
  timeout: 5m
```

Reply:
looks cleaner, let me update it

Reply:
I the proposed API I use `enable` for consistency with the analogous field in `InternalCertManagement`. Also, I suggest to use `timeoutSeconds` to express the timeout by seconds, similarly as Job's `ActiveDeadlineSeconds` is defined.